core/trino-main/src/main/java/io/trino/util/StatementUtils.java

```diff
@@ -183,18 +236,124 @@ private StatementUtils() {}
         return Stream.concat(
                 Stream.of(ExplainAnalyze.class),
                 STATEMENT_QUERY_TYPES.entrySet().stream()
-                        .filter(entry -> entry.getValue() != DATA_DEFINITION)
+                        .filter(entry -> entry.getValue().getQueryType() != DATA_DEFINITION)
                         .map(Map.Entry::getKey))
                 .collect(toImmutableSet());
     }
 
     public static boolean isDataDefinitionStatement(Class<? extends Statement> statement)
     {
-        return STATEMENT_QUERY_TYPES.get(statement) == DATA_DEFINITION;
+        StatementTypeInfo<? extends Statement> info = STATEMENT_QUERY_TYPES.get(statement);
+        return info != null && info.getQueryType() == DATA_DEFINITION;
+    }
+
+    public static <T extends Statement> Optional<String> explainDataDefinition(T statement, List<Expression> parameters)
+    {
+        if (!isDataDefinitionStatement(statement.getClass())) {
```

Comment:
Do we still need this as a separate method? It's only used in `QueryExecutionFactoryModule`, which seems like could be simplified now that we have the mapping here.